<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.226">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>SCpubr - News</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./99-Acknowledgements.html" rel="next">
<link href="./96-Citation.html" rel="prev">
<link href=".//images/SCpubr_logo_favicon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<meta property="og:title" content="SCpubr - News">
<meta property="og:description" content="Here are the patch notes of the different versions:">
<meta property="og:site-name" content="SCpubr">
<meta name="twitter:title" content="SCpubr - News">
<meta name="twitter:description" content="Here are the patch notes of the different versions:">
<meta name="twitter:card" content="summary">
</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">News</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">SCpubr </a><a href="https://github.com/enblacar/SCpubr/blob/main/NEWS.md"><code>v1.1.2</code></a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/enblacar/SCpubr/" title="Source Code" class="sidebar-tool px-1"><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">Getting started</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">About this package</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-Installation-guide.html" class="sidebar-item-text sidebar-link">Installation guide</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-Dataset.html" class="sidebar-item-text sidebar-link">Dataset used</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">Basic plots</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-DimPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Dim plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-FeaturePlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Feature plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-NebulosaPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Nebulosa plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-BeeSwarmPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Bee Swarm plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-ViolinPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Violin plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./08-RidgePlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Ridge plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./09-DotPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Dot plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10-BarPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Bar plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./11-BoxPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Box plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./12-GeyserPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Geyser plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13-AlluvialPlot.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Alluvial plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./13-SankeyPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Sankey plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./14-ChordDiagramPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Chord Diagram plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./15-VolcanoPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Volcano plots</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">Advanced plots</a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./16-GroupwiseDEPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">Group-wise DE analysis plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./17-GroupedGOPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Grouped GO Term analysis plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./17-FunctionalEnrichmentPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Functional Annotation Analysis plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./17-TermEnrichmentPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Term Enrichment Plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./18-ExpressionHeatmaps.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Expression heatmaps</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./18-EnrichmentHeatmaps.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Enrichment score heatmaps</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./19-CorrelationHeatmaps.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Correlation matrix heatmaps</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./20-CellularStatesPlot.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Cellular State Plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./21-LigandReceptorPlot.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Ligand-Receptor analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./22-CopyNumberVariantPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Copy Number Variant analysis plots</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./23-PathwayActivityPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Pathway Activity inference analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./24-TFActivityPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">TF Activity inference analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./25-AzimuthAnalysisPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Azimuth reference mapping reports</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./25-PseudotimeAnalysisPlots.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Pseudotime analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">Miscellaneous</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./93-ColorPalettes.html" class="sidebar-item-text sidebar-link">Color palettes</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./94-CommonFeatures.html" class="sidebar-item-text sidebar-link">Common features across plots</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./95-SaveFigures.html" class="sidebar-item-text sidebar-link">Save the figures</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./96-Citation.html" class="sidebar-item-text sidebar-link">Citation</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./97-NEWS.html" class="sidebar-item-text sidebar-link active">News</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./99-Acknowledgements.html" class="sidebar-item-text sidebar-link">Acknowledgements</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#scpubr-v1.1.1.9000" id="toc-scpubr-v1.1.1.9000" class="nav-link active" data-scroll-target="#scpubr-v1.1.1.9000">SCpubr v1.1.1.9000</a></li>
  <li><a href="#scpubr-v1.1.1" id="toc-scpubr-v1.1.1" class="nav-link" data-scroll-target="#scpubr-v1.1.1">SCpubr v1.1.1</a></li>
  <li><a href="#scpubr-v1.1.0" id="toc-scpubr-v1.1.0" class="nav-link" data-scroll-target="#scpubr-v1.1.0">SCpubr v1.1.0</a>
  <ul class="collapse">
  <li><a href="#general" id="toc-general" class="nav-link" data-scroll-target="#general">General</a></li>
  <li><a href="#scpubrdo_beeswarmplot" id="toc-scpubrdo_beeswarmplot" class="nav-link" data-scroll-target="#scpubrdo_beeswarmplot"><code>SCpubr::do_BeeSwarmPlot()</code></a></li>
  <li><a href="#scpubrdo_barplot" id="toc-scpubrdo_barplot" class="nav-link" data-scroll-target="#scpubrdo_barplot"><code>SCpubr::do_BarPlot()</code></a></li>
  <li><a href="#scpubrdo_boxplot" id="toc-scpubrdo_boxplot" class="nav-link" data-scroll-target="#scpubrdo_boxplot"><code>SCpubr::do_BoxPlot()</code></a></li>
  <li><a href="#scpubrdo_cellularstatesplot" id="toc-scpubrdo_cellularstatesplot" class="nav-link" data-scroll-target="#scpubrdo_cellularstatesplot"><code>SCpubr::do_CellularStatesPlot()</code></a></li>
  <li><a href="#scpubrdo_correlationplot" id="toc-scpubrdo_correlationplot" class="nav-link" data-scroll-target="#scpubrdo_correlationplot"><code>SCpubr::do_CorrelationPlot()</code></a></li>
  <li><a href="#scpubrdo_dimplot" id="toc-scpubrdo_dimplot" class="nav-link" data-scroll-target="#scpubrdo_dimplot"><code>SCpubr::do_DimPlot()</code></a></li>
  <li><a href="#scpubrdo_enrichmentheatmap" id="toc-scpubrdo_enrichmentheatmap" class="nav-link" data-scroll-target="#scpubrdo_enrichmentheatmap"><code>SCpubr::do_EnrichmentHeatmap()</code></a></li>
  <li><a href="#scpubrdo_featureplot" id="toc-scpubrdo_featureplot" class="nav-link" data-scroll-target="#scpubrdo_featureplot"><code>SCpubr::do_FeaturePlot()</code></a></li>
  <li><a href="#scpubrdo_groupwisedeplot" id="toc-scpubrdo_groupwisedeplot" class="nav-link" data-scroll-target="#scpubrdo_groupwisedeplot"><code>SCpubr::do_GroupwiseDEPlot()</code></a></li>
  <li><a href="#scpubrdo_ligandreceptorplot" id="toc-scpubrdo_ligandreceptorplot" class="nav-link" data-scroll-target="#scpubrdo_ligandreceptorplot"><code>SCpubr::do_LigandReceptorPlot()</code></a></li>
  <li><a href="#do_pathwayactivityplot" id="toc-do_pathwayactivityplot" class="nav-link" data-scroll-target="#do_pathwayactivityplot"><code>do_PathwayActivityPlot()</code></a></li>
  <li><a href="#do_tfactivityplot" id="toc-do_tfactivityplot" class="nav-link" data-scroll-target="#do_tfactivityplot"><code>do_TFActivityPlot()</code></a></li>
  <li><a href="#scpubrdo_violinplot" id="toc-scpubrdo_violinplot" class="nav-link" data-scroll-target="#scpubrdo_violinplot"><code>SCpubr::do_ViolinPlot()</code></a></li>
  </ul></li>
  <li><a href="#scpubr-v1.0.4" id="toc-scpubr-v1.0.4" class="nav-link" data-scroll-target="#scpubr-v1.0.4">SCpubr v1.0.4</a></li>
  <li><a href="#scpubr-v1.0.3" id="toc-scpubr-v1.0.3" class="nav-link" data-scroll-target="#scpubr-v1.0.3">SCpubr v1.0.3</a>
  <ul class="collapse">
  <li><a href="#general-changes" id="toc-general-changes" class="nav-link" data-scroll-target="#general-changes">General changes</a></li>
  <li><a href="#scpubrdo_dimplot-1" id="toc-scpubrdo_dimplot-1" class="nav-link" data-scroll-target="#scpubrdo_dimplot-1"><code>SCpubr::do_DimPlot()</code></a></li>
  <li><a href="#scpubrdo_enrichmentheatmap-1" id="toc-scpubrdo_enrichmentheatmap-1" class="nav-link" data-scroll-target="#scpubrdo_enrichmentheatmap-1"><code>SCpubr::do_EnrichmentHeatmap()</code></a></li>
  <li><a href="#scpubrdo_featureplot-1" id="toc-scpubrdo_featureplot-1" class="nav-link" data-scroll-target="#scpubrdo_featureplot-1"><code>SCpubr::do_FeaturePlot()</code></a></li>
  <li><a href="#scpubrdo_geyserplot" id="toc-scpubrdo_geyserplot" class="nav-link" data-scroll-target="#scpubrdo_geyserplot"><code>SCpubr::do_GeyserPlot()</code></a></li>
  <li><a href="#scpubrdo_ridgeplot" id="toc-scpubrdo_ridgeplot" class="nav-link" data-scroll-target="#scpubrdo_ridgeplot"><code>SCpubr::do_RidgePlot()</code></a></li>
  <li><a href="#scpubrdo_violinplot-1" id="toc-scpubrdo_violinplot-1" class="nav-link" data-scroll-target="#scpubrdo_violinplot-1"><code>SCpubr::do_ViolinPlot()</code></a></li>
  </ul></li>
  <li><a href="#scpubr-v1.0.2" id="toc-scpubr-v1.0.2" class="nav-link" data-scroll-target="#scpubr-v1.0.2">SCpubr v1.0.2</a>
  <ul class="collapse">
  <li><a href="#general-changes-1" id="toc-general-changes-1" class="nav-link" data-scroll-target="#general-changes-1">General changes</a></li>
  <li><a href="#scpubrdo_copynumbervariantplot" id="toc-scpubrdo_copynumbervariantplot" class="nav-link" data-scroll-target="#scpubrdo_copynumbervariantplot"><code>SCpubr::do_CopyNumberVariantPlot</code></a></li>
  <li><a href="#scpubrdo_dimplot-2" id="toc-scpubrdo_dimplot-2" class="nav-link" data-scroll-target="#scpubrdo_dimplot-2"><code>SCpubr::do_DimPlot()</code></a></li>
  <li><a href="#scpubrdo_dotplot" id="toc-scpubrdo_dotplot" class="nav-link" data-scroll-target="#scpubrdo_dotplot"><code>SCpubr::do_DotPlot()</code></a></li>
  <li><a href="#scpubrdo_featureplot-2" id="toc-scpubrdo_featureplot-2" class="nav-link" data-scroll-target="#scpubrdo_featureplot-2"><code>SCpubr::do_FeaturePlot()</code></a></li>
  <li><a href="#scpubrdo_geyserplot-1" id="toc-scpubrdo_geyserplot-1" class="nav-link" data-scroll-target="#scpubrdo_geyserplot-1"><code>SCpubr::do_GeyserPlot()</code></a></li>
  <li><a href="#scpubrdo_groupwisedeplot-1" id="toc-scpubrdo_groupwisedeplot-1" class="nav-link" data-scroll-target="#scpubrdo_groupwisedeplot-1"><code>SCpubr::do_GroupwiseDEPlot()</code></a></li>
  <li><a href="#scpubrdo_ligandreceptorplot-development-release" id="toc-scpubrdo_ligandreceptorplot-development-release" class="nav-link" data-scroll-target="#scpubrdo_ligandreceptorplot-development-release"><code>SCpubr::do_LigandReceptorPlot()</code> (development release)</a></li>
  <li><a href="#scpubrdo_ridgeplot-1" id="toc-scpubrdo_ridgeplot-1" class="nav-link" data-scroll-target="#scpubrdo_ridgeplot-1"><code>SCpubr::do_RidgePlot()</code></a></li>
  <li><a href="#scpubrdo_violinplot-2" id="toc-scpubrdo_violinplot-2" class="nav-link" data-scroll-target="#scpubrdo_violinplot-2"><code>SCpubr::do_ViolinPlot()</code></a></li>
  <li><a href="#scpubr-v1.0.2-dev-stable" id="toc-scpubr-v1.0.2-dev-stable" class="nav-link" data-scroll-target="#scpubr-v1.0.2-dev-stable">SCpubr v1.0.2-dev-stable</a></li>
  </ul></li>
  <li><a href="#scpubr-v1.0.1" id="toc-scpubr-v1.0.1" class="nav-link" data-scroll-target="#scpubr-v1.0.1">SCpubr v1.0.1</a>
  <ul class="collapse">
  <li><a href="#scpubr-v1.0.1-dev-stable" id="toc-scpubr-v1.0.1-dev-stable" class="nav-link" data-scroll-target="#scpubr-v1.0.1-dev-stable">SCpubr v1.0.1-dev-stable</a></li>
  </ul></li>
  <li><a href="#scpubr-v1.0.0" id="toc-scpubr-v1.0.0" class="nav-link" data-scroll-target="#scpubr-v1.0.0">SCpubr v1.0.0</a>
  <ul class="collapse">
  <li><a href="#scpubr-v1.0.0-dev-stable" id="toc-scpubr-v1.0.0-dev-stable" class="nav-link" data-scroll-target="#scpubr-v1.0.0-dev-stable">SCpubr v1.0.0-dev-stable</a></li>
  </ul></li>
  <li><a href="#scpubr-0.0.0.9000" id="toc-scpubr-0.0.0.9000" class="nav-link" data-scroll-target="#scpubr-0.0.0.9000">SCpubr 0.0.0.9000</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/enblacar/SCpubr/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">News</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>Here are the patch notes of the different versions:</p>
<section id="scpubr-v1.1.1.9000" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.1.1.9000">SCpubr v1.1.1.9000</h2>
<p>Current development release.</p>
</section>
<section id="scpubr-v1.1.1" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.1.1">SCpubr v1.1.1</h2>
<p>Hotfixes in unit tests to comply with CRAN checks.</p>
</section>
<section id="scpubr-v1.1.0" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.1.0">SCpubr v1.1.0</h2>
<section id="general" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="general">General</h3>
<ul>
<li>Increased the cell size of all heatmap-based functions from 5 to 8.</li>
<li>Decreased the thickness of frame and ticks of all ggplot2-based continuous legends to retrieve a similar behavior as in previous versions of ggplot2, as with the new update, the overall thickness of the frame and ticks increased, probably due to the changes related to <code>element_line</code>,</li>
<li>Added five new functions: <code>do_AlluvialPlot()</code>, <code>do_AzimuthAnalysisPlot()</code>, <code>do_ExpressionHeatmap()</code>, <code>do_GroupedGOTermPlot()</code> and <code>do_FunctionalAnnotationPlot()</code>.</li>
<li>Added <code>legend.ncol</code>, <code>legend.nrow</code>, <code>legend.title</code> and <code>legend.byrow</code> to as many functions as possible to further customize legend appearance.</li>
</ul>
</section>
<section id="scpubrdo_beeswarmplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_beeswarmplot"><code>SCpubr::do_BeeSwarmPlot()</code></h3>
<ul>
<li>Added <code>min.cutoff</code> and <code>max.cutoff</code> parameter.</li>
<li>Added ticks to the plot, that were missing.</li>
<li>Added missing axes titles.</li>
<li>Added <code>viridis_direction</code> parameter to control how the continuous color scale is formed.</li>
<li>Added <code>return_object</code> parameter to return the Seurat object with the enrichment scores computed.</li>
<li>Added BoxPlots, BeeSwarmPlots and ViolinPlots to the possible outputs the user can choose from.</li>
<li>Make <code>legend.position</code> conditional of whether <code>continuous_feature</code> is set to TRUE. If it is false, legend is not displayed unless the user specifies otherwise.</li>
</ul>
</section>
<section id="scpubrdo_barplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_barplot"><code>SCpubr::do_BarPlot()</code></h3>
<ul>
<li>Fixed a bug in which axes titles were not displaying correctly under certain combinations of <code>flip</code> and <code>split.by</code>.</li>
<li>Fixed a bug in which <code>x_lab</code> and <code>y_lab</code> would not rotate accordingly when using <code>flip = TRUE</code>.</li>
</ul>
</section>
<section id="scpubrdo_boxplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_boxplot"><code>SCpubr::do_BoxPlot()</code></h3>
<ul>
<li>Set <code>assay</code> to NULL and will default to the default assay in the seurat object.</li>
</ul>
</section>
<section id="scpubrdo_cellularstatesplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_cellularstatesplot"><code>SCpubr::do_CellularStatesPlot()</code></h3>
<ul>
<li>Fixed a bug that prevented FeaturePlots to have symmetrical axes with respect to the main plot.</li>
</ul>
</section>
<section id="scpubrdo_correlationplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_correlationplot"><code>SCpubr::do_CorrelationPlot()</code></h3>
<ul>
<li>Added <code>viridis_direction</code> parameter.</li>
</ul>
</section>
<section id="scpubrdo_dimplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_dimplot"><code>SCpubr::do_DimPlot()</code></h3>
<ul>
<li>Fixed a bug in which the legend title will not show up in regular basic plots even though the parameter <code>legend.title</code> was used.</li>
<li>Completely reformatted the way <code>split.by</code> works, so that now only one legend is displayed for the whole group and cells have border.</li>
<li>Added <code>label.size</code> and <code>label.box</code> parameters for further customize the appearance of the plot when using <code>label = TRUE</code>.</li>
<li>Changed <code>repel</code> to <code>FALSE</code> by default.</li>
</ul>
</section>
<section id="scpubrdo_enrichmentheatmap" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_enrichmentheatmap"><code>SCpubr::do_EnrichmentHeatmap()</code></h3>
<ul>
<li>Fixed a bug in the code that prevented the feature plots and the geyser plots to be computed if the input gene list was not a named list of genes.</li>
<li>Added <code>flavor = "AUCell"</code>, that lets the user compute AUCell scoring of the gene sets.</li>
<li>Added the option to query multiple <code>group.by</code> parameters at the same time.</li>
<li>Fixed a bug in the code that prevented multiple outputs with different values of <code>group.by</code> to be returned properly, leading to the last value of <code>group.by</code> replacing all the rest.</li>
</ul>
</section>
<section id="scpubrdo_featureplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_featureplot"><code>SCpubr::do_FeaturePlot()</code></h3>
<ul>
<li>Added <code>label</code>, <code>label.size</code> and <code>label.color</code> parameter to reproduce the same behavior as in <code>Seurat::FeaturePlot()</code>.</li>
</ul>
</section>
<section id="scpubrdo_groupwisedeplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_groupwisedeplot"><code>SCpubr::do_GroupwiseDEPlot()</code></h3>
<ul>
<li>Set <code>assay</code> to NULL and will default to the default assay in the seurat object.</li>
</ul>
</section>
<section id="scpubrdo_ligandreceptorplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_ligandreceptorplot"><code>SCpubr::do_LigandReceptorPlot()</code></h3>
<ul>
<li>Added <code>arrange_interactions_by</code> to control how output interactions are arranged (either by aggregate_rank, specificity, magnitude or a combination of magnitude and specificity).</li>
<li>Added <code>sort_interactions_alphabetically</code> to control whether the output dotplot has the interactions ordered alphabetically or as they come in the original matrix (meaning, they follow the arrangement specified in <code>arrange_interactions_by</code>). ((<a href="https://github.com/saezlab/liana/issues/72">liana’s issue #72</a>))</li>
</ul>
</section>
<section id="do_pathwayactivityplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="do_pathwayactivityplot"><code>do_PathwayActivityPlot()</code></h3>
<ul>
<li>Added a fix in which when <code>enforce_symmetry</code> is set to <code>FALSE</code>, then the color scale turns into a viridis-based one instead of a two-color gradient scale.</li>
</ul>
</section>
<section id="do_tfactivityplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="do_tfactivityplot"><code>do_TFActivityPlot()</code></h3>
<ul>
<li>Added a fix in which when <code>enforce_symmetry</code> is set to <code>FALSE</code>, then the color scale turns into a viridis-based one instead of a two-color gradient scale.</li>
</ul>
</section>
<section id="scpubrdo_violinplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_violinplot"><code>SCpubr::do_ViolinPlot()</code></h3>
<ul>
<li>Fixed a bug in the code in which no different colors could be passed to <code>colors.use</code>.</li>
<li>Reduced default line width from 1 to 0.5.</li>
</ul>
</section>
</section>
<section id="scpubr-v1.0.4" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.4">SCpubr v1.0.4</h2>
<ul>
<li>Hotfix for v1.0.3 in which <code>do_GeyserPlot</code> with categorical variables had a bug that mapped the legend to the continuous axis.</li>
</ul>
</section>
<section id="scpubr-v1.0.3" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.3">SCpubr v1.0.3</h2>
<section id="general-changes" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="general-changes">General changes</h3>
<ul>
<li>Added <code>min.cutoff</code> and <code>max.cutoff</code> parameter to effectively subset the color scale and remove the effect of extreme outliers in all ComplexHeatmap-based functions.</li>
<li>Added <code>min.cutoff</code> and <code>max.cutoff</code> parameter to effectively subset the color scale and remove the effect of extreme outliers in all ggplot2-based functions susceptible to be biased by outliers.</li>
</ul>
</section>
<section id="scpubrdo_dimplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_dimplot-1"><code>SCpubr::do_DimPlot()</code></h3>
<ul>
<li>Implemented a change in which when using <code>split.by</code> and <code>group.by</code> in combination, the cells colored on top of the UMAP also have a border.</li>
<li>Implemented a bug-fix in which when using <code>split.by</code> and <code>group.by</code> in combination, the extra new layers would not raster if <code>raster = TRUE</code>.</li>
<li>Implemented a bug-fix in which when using <code>split.by</code> and <code>group.by</code> in combination, no plots will appear if <code>ncol</code> is set.</li>
<li>Implemented a new feature to add density line contours using <code>plot_density_contour</code>.</li>
<li>Implemented the conditional use of <code>raster.dpi</code> to Seurat versions higher or equal to 4.1.0.</li>
</ul>
</section>
<section id="scpubrdo_enrichmentheatmap-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_enrichmentheatmap-1"><code>SCpubr::do_EnrichmentHeatmap()</code></h3>
<ul>
<li>Implemented a bug fix for internal checks in the function.</li>
<li>Added <code>plot_FeaturePlots</code> and <code>plot_GeyserPlots</code> to also report the enrichment scores in a gene set-based manner.</li>
<li>Added <code>flavor</code> parameter, that accepts <code>Seurat</code> and <code>UCell</code> to allow for different enrichment scoring methods. It requires <code>R 4.2.0</code> to run.</li>
<li>Renamed <code>symmetrical_scale</code> to <code>enforce_symmetry</code> to have a greater coherence accross functions.</li>
</ul>
</section>
<section id="scpubrdo_featureplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_featureplot-1"><code>SCpubr::do_FeaturePlot()</code></h3>
<ul>
<li>Implemented a new feature to add density line contours using <code>plot_density_contour</code>.</li>
<li>Implemented the conditional use of <code>raster.dpi</code> to Seurat versions higher or equal to 4.1.0.</li>
</ul>
</section>
<section id="scpubrdo_geyserplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_geyserplot"><code>SCpubr::do_GeyserPlot()</code></h3>
<ul>
<li>Fixed bug in which internal parameter names made it to the X axis title.</li>
<li>Removed <code>color.by</code> implementation due to it being very buggy. This will be re-implemented in a future patch.</li>
</ul>
</section>
<section id="scpubrdo_ridgeplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_ridgeplot"><code>SCpubr::do_RidgePlot()</code></h3>
<ul>
<li>Implemented a bug-fix in which using <code>assay = "RNA"</code> or, in fact, any other assay rather than <code>SCT</code> will result in an error.</li>
</ul>
</section>
<section id="scpubrdo_violinplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_violinplot-1"><code>SCpubr::do_ViolinPlot()</code></h3>
<ul>
<li>Corrected a bug in which legend title when using <code>split.by</code> was an actual line of code.</li>
<li>Added <code>legend.title</code> parameter to control the title of the legend.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v1.0.3">v1.0.3 release</a> release or <strong>downloading it from CRAN</strong> using:</p>
<section id="scpubr-v.1.0.3-dev-stable" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="scpubr-v.1.0.3-dev-stable">SCpubr v.1.0.3-dev-stable</h4>
<ul>
<li>Same as v1.0.3, but with all the functions that do not pass CRAN checks. These functions are: <code>SCpubr::save_Plot()</code> <code>SCpubr::do_LigandReceptorPlot()</code> and <code>SCpubr::do_SankeyPlot()</code>.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v.1.0.3-dev-stable">v1.0.3-dev-stable release</a> release.</p>
</section>
</section>
</section>
<section id="scpubr-v1.0.2" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.2">SCpubr v1.0.2</h2>
<section id="general-changes-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="general-changes-1">General changes</h3>
<ul>
<li>Change color palette when using <code>enforce_symmetry = TRUE</code> to have the middle color as <code>grey95</code> instead of the previous one, which made middle values seem closer to the positive end of the scale.</li>
<li>Modified internal structure of all functions to abide with <a href="https://tidyselect.r-lib.org/news/index.html#lifecycle-changes-1-2-0">tidyselect v1.2.0 lifecycle modifications</a>.</li>
<li>Modified <code>rotate_x_axis_labels</code> parameter in all functions that made use of it. Now, instead of accepting a <code>logical</code>, accepts a <code>numeric</code>: either <code>0</code>, <code>45</code> or <code>90</code>, corresponding to the degrees in which the X axis labels should be rotated. (<a href="https://github.com/enblacar/SCpubr/issues/5#issuecomment-1289203453">#5</a>)</li>
</ul>
</section>
<section id="scpubrdo_copynumbervariantplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_copynumbervariantplot"><code>SCpubr::do_CopyNumberVariantPlot</code></h3>
<ul>
<li>Modified the code for <code>SCpubr::do_CopyNumberVariantPlot</code> to also report results for the whole chromosome as well as for each chromosome arm.</li>
<li>Include the <code>verbose</code> argument to <code>SCpubr::do_CopyNumberVariantPlot</code> to silence the messages when there are not enough genes in the chromosome to perform the analysis.</li>
</ul>
</section>
<section id="scpubrdo_dimplot-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_dimplot-2"><code>SCpubr::do_DimPlot()</code></h3>
<ul>
<li>Fixed a typo that prevented labels to be bold in <code>SCpubr::do_DimPlot()</code> when cell borders are displayed.</li>
<li>Added <code>group.by</code> and <code>split.by</code> functionality to <code>SCpubr::do_DimPlot()</code>. (<a href="https://github.com/enblacar/SCpubr/issues/4">#4</a>)</li>
</ul>
</section>
<section id="scpubrdo_dotplot" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_dotplot"><code>SCpubr::do_DotPlot()</code></h3>
<ul>
<li>Added ticks to axes.</li>
<li>Modified default colors to convey a better aesthetic.</li>
</ul>
</section>
<section id="scpubrdo_featureplot-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_featureplot-2"><code>SCpubr::do_FeaturePlot()</code></h3>
<ul>
<li>Fixed potential bugs in <code>SCpubr::do_FeaturePlot</code> when setting <code>enforce_symmetry = TRUE</code>.</li>
<li>Changed default value of <code>order</code> in <code>SCpubr::do_FeaturePlot()</code> from <code>TRUE</code> to <code>FALSE</code>.</li>
<li>Added <code>min.cutoff</code> and <code>max.cutoff</code> parameters to <code>SCpubr::do_FeaturePlot()</code>. This allows to effectively subset the color scale to the values provided. Cells outside the range will be converted to the min or max values provided, respectively. (<a href="https://github.com/enblacar/SCpubr/issues/2">#2</a>)</li>
</ul>
</section>
<section id="scpubrdo_geyserplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_geyserplot-1"><code>SCpubr::do_GeyserPlot()</code></h3>
<ul>
<li>Added <code>flip</code> parameter.</li>
</ul>
</section>
<section id="scpubrdo_groupwisedeplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_groupwisedeplot-1"><code>SCpubr::do_GroupwiseDEPlot()</code></h3>
<ul>
<li>Fixed bug in <code>SCpubr::do_GroupwiseDEPlot</code> in which the heatmap could not be computed. (<a href="https://github.com/enblacar/SCpubr/issues/3">#3</a>)</li>
<li>Added extra checks to ensure proper input in <code>SCpubr::do_GroupwiseDEPlot</code>. (<a href="https://github.com/enblacar/SCpubr/issues/3">#3</a>)</li>
</ul>
</section>
<section id="scpubrdo_ligandreceptorplot-development-release" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_ligandreceptorplot-development-release"><code>SCpubr::do_LigandReceptorPlot()</code> (development release)</h3>
<ul>
<li>Changed parameter <code>x_labels_angle</code> to <code>rotate_x_axis_labels</code> to keep a consistent terminology.</li>
</ul>
</section>
<section id="scpubrdo_ridgeplot-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_ridgeplot-1"><code>SCpubr::do_RidgePlot()</code></h3>
<ul>
<li>Fixed a typo that made the lines in <code>panel.grid.minor</code> to be displayed in <code>SCpubr::do_Ridgeplot()</code>.</li>
<li>Added <code>flip</code> parameter.</li>
</ul>
</section>
<section id="scpubrdo_violinplot-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubrdo_violinplot-2"><code>SCpubr::do_ViolinPlot()</code></h3>
<ul>
<li>Added <code>split.by</code> functionality to <code>SCpubr::do_ViolinPlot()</code>. (<a href="https://github.com/enblacar/SCpubr/issues/4">#4</a>, <a href="https://github.com/enblacar/SCpubr/issues/5">#5</a>)</li>
<li>Added <code>flip</code> parameter.</li>
<li>Now multiple features can be queried ad the same time. (<a href="https://github.com/enblacar/SCpubr/issues/5#issuecomment-1289203453">#5</a>)</li>
<li>Changed <code>feature</code> parameter to <code>features</code>, to better reflect the multiple feature behavior.</li>
<li>Recreated <code>Seurat</code>’s <code>share.y.lims</code> behavior and set it to <code>share.y.lims</code> parameter. (<a href="https://github.com/enblacar/SCpubr/issues/5#issuecomment-1289203453">#5</a>)</li>
</ul>
</section>
<section id="scpubr-v1.0.2-dev-stable" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.2-dev-stable">SCpubr v1.0.2-dev-stable</h3>
<ul>
<li>Same as v1.0.2, but with all the functions that do not pass CRAN checks. These functions are: <code>SCpubr::save_Plot()</code> <code>SCpubr::do_LigandReceptorPlot()</code> and <code>SCpubr::do_SankeyPlot()</code>.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v1.0.2-dev-stable">v1.0.2-dev-stable release</a> release.</p>
</section>
</section>
<section id="scpubr-v1.0.1" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.1">SCpubr v1.0.1</h2>
<ul>
<li>Rework on unit tests and examples so that it can pass CRAN R CMD Check without packages in Suggests. This is, to make sure all Suggested packages are used conditionally.</li>
</ul>
<section id="scpubr-v1.0.1-dev-stable" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.1-dev-stable">SCpubr v1.0.1-dev-stable</h3>
<ul>
<li>Same as v1.0.1, but with all the functions that do not pass CRAN checks. These functions are: <code>SCpubr::save_Plot()</code> <code>SCpubr::do_LigandReceptorPlot()</code> and <code>SCpubr::do_SankeyPlot()</code>.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v1.0.1-dev-stable">v1.0.1-dev-stable release</a> release.</p>
</section>
</section>
<section id="scpubr-v1.0.0" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.0">SCpubr v1.0.0</h2>
<ul>
<li>Modified internal checks so that the functions that do not use <code>Seurat</code> do not require this to be installed. This is just for the very side case in which somebody downloads the package just for the <code>SCpubr::do_ColorPalette()</code> function.</li>
<li>Removed the option to use <code>individual.titles</code>, ’individual.subtitles<code>and</code>individual.captions<code>in</code>SCpubr::do_NebulosaPlot()` as the benefit of such parameters did not surpass the problems the code was causing. The feature might come back in the future, once fully optimized.</li>
<li>Removed <code>SCpubr::save_Plot()</code> function to align with CRAN policies that the package should not write to the file system. The code is still available in the v0.0.0.0.9000 release in Github.</li>
<li>Removed <code>SCpubr::do_LigandReceptorPlot()</code>, <code>SCpubr::do_SankeyPlot()</code> and <code>SCpubr::do_PseudotimePlot()</code> to align with CRAN policies and make it possible to publish the package. These functions can still be accessed in the v0.0.0.0.9000 release in Github.</li>
<li>Removed <code>SCpubr::do_PseudotimePlot()</code> for the reason above and because the dependency <code>Matrix.utils</code> was removed from CRAN on <em>09-10-2022</em>.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v1.0.0">v1.0.0 release</a> release.</p>
<section id="scpubr-v1.0.0-dev-stable" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="scpubr-v1.0.0-dev-stable">SCpubr v1.0.0-dev-stable</h3>
<ul>
<li>Same as v1.0.0, but with all the functions that do not pass CRAN checks. These functions are: <code>SCpubr::save_Plot()</code> <code>SCpubr::do_LigandReceptorPlot()</code> and <code>SCpubr::do_SankeyPlot()</code>.</li>
</ul>
<p>This version can be obtained in the <a href="https://github.com/enblacar/SCpubr/releases/tag/v1.0.0-dev-stable">v1.0.0-dev-stable release</a> release.</p>
</section>
</section>
<section id="scpubr-0.0.0.9000" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="scpubr-0.0.0.9000">SCpubr 0.0.0.9000</h2>
<ul>
<li>Added a <code>NEWS.md</code> file to track changes to the package.</li>
<li>Prepare package for submission to CRAN.</li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./96-Citation.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Citation</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./99-Acknowledgements.html" class="pagination-link">
        <span class="nav-page-text">Acknowledgements</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">Written by <strong>Enrique Blanco Carmona</strong></div>   
    <div class="nav-footer-right">This book was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>