<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>16 Copy Number Variant analysis plots | SCpubr</title>
<meta name="author" content="Enrique Blanco Carmona">
<meta name="description" content="Copy Number Variant analysis are another of the common analysis that one can compute on single-cell transcriptomics data. Provided with a reference one can compute, for the rest of the cells,...">
<meta name="generator" content="bookdown 0.28 with bs4_book()">
<meta property="og:title" content="16 Copy Number Variant analysis plots | SCpubr">
<meta property="og:type" content="book">
<meta property="og:url" content="https://enblacar.github.io/SCpubr-book/copy-number-variant-analysis-plots.html">
<meta property="og:image" content="https://enblacar.github.io/SCpubr-book/images/logo.png">
<meta property="og:description" content="Copy Number Variant analysis are another of the common analysis that one can compute on single-cell transcriptomics data. Provided with a reference one can compute, for the rest of the cells,...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="16 Copy Number Variant analysis plots | SCpubr">
<meta name="twitter:description" content="Copy Number Variant analysis are another of the common analysis that one can compute on single-cell transcriptomics data. Provided with a reference one can compute, for the rest of the cells,...">
<meta name="twitter:image" content="https://enblacar.github.io/SCpubr-book/images/logo.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.0/transition.js"></script><script src="libs/bs3compat-0.4.0/tabs.js"></script><script src="libs/bs3compat-0.4.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link rel="shortcut icon" href="images/SCpubr_logo_favicon.png">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a id="toc_title" href="index.html" title="">SCpubr</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <div class="book-extra"></div>
      <nav id="left-panel" aria-label="Table of contents"><ul class="book-toc list-unstyled">
<li><a class="" href="index.html">About this package</a></li>
<li><a class="" href="installation-guide.html">Installation guide</a></li>
<li><a class="" href="dataset-used.html">Dataset used</a></li>
<li><a class="" href="dim-plots.html"><span class="header-section-number">1</span> Dim plots</a></li>
<li><a class="" href="feature-plots.html"><span class="header-section-number">2</span> Feature plots</a></li>
<li><a class="" href="nebulosa-plots.html"><span class="header-section-number">3</span> Nebulosa plots</a></li>
<li><a class="" href="bee-swarm-plots.html"><span class="header-section-number">4</span> Bee Swarm plots</a></li>
<li><a class="" href="violin-plots.html"><span class="header-section-number">5</span> Violin plots</a></li>
<li><a class="" href="ridge-plots.html"><span class="header-section-number">6</span> Ridge plots</a></li>
<li><a class="" href="dot-plots.html"><span class="header-section-number">7</span> Dot plots</a></li>
<li><a class="" href="bar-plots.html"><span class="header-section-number">8</span> Bar plots</a></li>
<li><a class="" href="geyser-plots.html"><span class="header-section-number">9</span> Geyser plots</a></li>
<li><a class="" href="group-wise-de-analysis-plots.html"><span class="header-section-number">10</span> Group-wise DE analysis plots</a></li>
<li><a class="" href="term-enrichment-plots.html"><span class="header-section-number">11</span> Term Enrichment Plots</a></li>
<li><a class="" href="enrichment-score-heatmaps.html"><span class="header-section-number">12</span> Enrichment score heatmaps</a></li>
<li><a class="" href="correlation-matrix-heatmaps.html"><span class="header-section-number">13</span> Correlation matrix heatmaps</a></li>
<li><a class="" href="cellular-state-plots.html"><span class="header-section-number">14</span> Cellular State Plots</a></li>
<li><a class="" href="ligand-receptor-analysis.html"><span class="header-section-number">15</span> Ligand-Receptor analysis</a></li>
<li><a class="active" href="copy-number-variant-analysis-plots.html"><span class="header-section-number">16</span> Copy Number Variant analysis plots</a></li>
<li><a class="" href="pathway-activity-inference-analysis.html"><span class="header-section-number">17</span> Pathway Activity inference analysis</a></li>
<li><a class="" href="tf-activity-inference-analysis.html"><span class="header-section-number">18</span> TF Activity inference analysis</a></li>
<li><a class="" href="pseudotime-analysis.html"><span class="header-section-number">19</span> Pseudotime analysis</a></li>
<li><a class="" href="color-palettes.html"><span class="header-section-number">20</span> Color palettes</a></li>
<li><a class="" href="common-features-across-plots.html">Common features across plots</a></li>
<li><a class="" href="rasterized-figures.html">Rasterized figures</a></li>
<li><a class="" href="save-the-figures.html">Save the figures</a></li>
<li><a class="" href="troubleshooting.html">Troubleshooting</a></li>
<li><a class="" href="citation.html">Citation</a></li>
</ul>

        <div class="book-extra">
          <!-- Duplicated this entry to point to the package repo. -->
          <p class="mb-2"><a id="package-repo" href="https://github.com/enblacar/SCpubr">View package source <i class="fab fa-github"></i></a></p>
          <p><a id="book-repo" href="https://github.com/enblacar/SCpubr-book">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="copy-number-variant-analysis-plots" class="section level1" number="16">
<h1>
<span class="header-section-number">16</span> Copy Number Variant analysis plots<a class="anchor" aria-label="anchor" href="#copy-number-variant-analysis-plots"><i class="fas fa-link"></i></a>
</h1>
<p>Copy Number Variant analysis are another of the common analysis that one can compute on single-cell transcriptomics data. Provided with a reference one can compute, for the rest of the cells, whether there are Copy Number Variations (CNVs) in the cells across the chromosomes. This comes really handy to distinguish between tumor and healthy cells, provided that one has a CNV reference event to rely on. There are many tools to compute such analysis, but one that is highly used and that will be covered in this section is <a href="https://github.com/broadinstitute/infercnv">inferCNV</a>.</p>
<p>InferCNV returns many outputs. The most interesting and straightforward one is the final image, such as the one below:
<span class="border-0"><img src="images/infercnv.png" class="mx-auto d-block" alt="" style="box-shadow: none; width: 100%"></span></p>
<p>In it, we can observe that, for the different clusters, regions called as chromosome gains are colored as red and regions called as chromosome losses are colored as blue. This image, in the end, is a heatmap, meaning that there is, in the output of inferCNV, a matrix with the scores for each cell and gene, that we can make use of. This object is called, by default, <em>run.final.infercnv_obj</em>. This is the one that we will use.</p>
<div id="transferring-the-scores-to-a-featureplot" class="section level2" number="16.1">
<h2>
<span class="header-section-number">16.1</span> Transferring the scores to a FeaturePlot<a class="anchor" aria-label="anchor" href="#transferring-the-scores-to-a-featureplot"><i class="fas fa-link"></i></a>
</h2>
<p>One of the cool things we can do with this object, is to transfer the inferCNV scores back to our Seurat object and then plot them as a FeaturePlot. This can be achieved with the function <code><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">SCpubr::do_CopyNumberVariantPlot()</a></code>. For this function, one needs to provide the Seurat object and the final inferCNV object, together with the chromosome locations. If metacells were computed (not necessary, but used in this example), the mapping of cell-metacell has to be provided as well:</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This loads "human_chr_locations" into the environment.</span></span>
<span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"human_chr_locations"</span>, package <span class="op">=</span> <span class="st">"SCpubr"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">do_CopyNumberVariantPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                                        infercnv_object <span class="op">=</span> <span class="va">infercnv_object</span>,</span>
<span>                                        using_metacells <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        metacell_mapping <span class="op">=</span> <span class="va">sample</span><span class="op">$</span><span class="va">metacell_mapping</span>,</span>
<span>                                        chromosome_locations <span class="op">=</span> <span class="va">human_chr_locations</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">`11p_umap`</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-2"></span>
<img src="21-CopyNumberVariantPlots_files/figure-html/unnamed-chunk-2-1.png" alt="SCpubr::do_CopyNumberVariantPlot FeaturePlot." width="100%" height="100%"><p class="caption">
Figure 1.1: SCpubr::do_CopyNumberVariantPlot FeaturePlot.
</p>
</div>
<p>Scores of 1 mean no chromosome gain or loss. Higher than one mean gain and lower mean loss. The function automatically computes the plots for <em>all chromosome regions</em>. If we want to restrict the output to a single chromosome, we can do so by stating it in <code>chromosome_focus</code> parameter.</p>
<div class="sourceCode" id="cb138"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">do_CopyNumberVariantPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                                        infercnv_object <span class="op">=</span> <span class="va">infercnv_object</span>,</span>
<span>                                        using_metacells <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        metacell_mapping <span class="op">=</span> <span class="va">sample</span><span class="op">$</span><span class="va">metacell_mapping</span>,</span>
<span>                                        chromosome_locations <span class="op">=</span> <span class="va">human_chr_locations</span>,</span>
<span>                                        chromosome_focus <span class="op">=</span> <span class="st">"11"</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="plotting-the-scores-grouped-by-a-group-of-interest" class="section level2" number="16.2">
<h2>
<span class="header-section-number">16.2</span> Plotting the scores grouped by a group of interest<a class="anchor" aria-label="anchor" href="#plotting-the-scores-grouped-by-a-group-of-interest"><i class="fas fa-link"></i></a>
</h2>
<p>Sometimes, however, the FeaturePlot is not sufficient to get the insights we want from these scores. For this, the output of <code><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">SCpubr::do_CopyNumberVariantPlot()</a></code> also contains another set of Geyser plots, that showcase the distribution of scores per cells, grouped by a variable of interest.</p>
<div class="sourceCode" id="cb139"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">do_CopyNumberVariantPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                                        infercnv_object <span class="op">=</span> <span class="va">infercnv_object</span>,</span>
<span>                                        using_metacells <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        metacell_mapping <span class="op">=</span> <span class="va">sample</span><span class="op">$</span><span class="va">metacell_mapping</span>,</span>
<span>                                        chromosome_locations <span class="op">=</span> <span class="va">human_chr_locations</span>,</span>
<span>                                        chromosome_focus <span class="op">=</span> <span class="st">"11"</span>,</span>
<span>                                        rotate_x_axis_labels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">`11p_geyser`</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="21-CopyNumberVariantPlots_files/figure-html/unnamed-chunk-4-1.png" alt="SCpubr::do_CopyNumberVariantPlot dot plot." width="100%" height="100%"><p class="caption">
Figure 1.3: SCpubr::do_CopyNumberVariantPlot dot plot.
</p>
</div>
<p>Here, we can observe the scores for the different groups, being each dot a cells. Due to the overplotting, we also report the distribution of the data for each group in the center, being the dot the median of the distribution, the thicker lines representing the 66% of the data and the thinner lines the 95%. This way, one can also see where the majority of the cells reside in each group. We can also select other variables to group by.</p>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample</span><span class="op">$</span><span class="va">modified_orig.ident</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sample_A"</span>, <span class="st">"Sample_B"</span>, <span class="st">"Sample_C"</span><span class="op">)</span>, </span>
<span>                                     size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span>, </span>
<span>                                     replace <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                                     prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.7</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">do_CopyNumberVariantPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                                        infercnv_object <span class="op">=</span> <span class="va">infercnv_object</span>,</span>
<span>                                        using_metacells <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        group.by <span class="op">=</span> <span class="st">"modified_orig.ident"</span>,</span>
<span>                                        metacell_mapping <span class="op">=</span> <span class="va">sample</span><span class="op">$</span><span class="va">metacell_mapping</span>,</span>
<span>                                        chromosome_locations <span class="op">=</span> <span class="va">human_chr_locations</span>,</span>
<span>                                        chromosome_focus <span class="op">=</span> <span class="st">"11"</span>,</span>
<span>                                        rotate_x_axis_labels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">`11p_geyser`</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="21-CopyNumberVariantPlots_files/figure-html/unnamed-chunk-5-1.png" alt="SCpubr::do_CopyNumberVariantPlot dot plot." width="100%" height="100%"><p class="caption">
Figure 1.4: SCpubr::do_CopyNumberVariantPlot dot plot.
</p>
</div>
</div>
<div id="joint-analysis" class="section level2" number="16.3">
<h2>
<span class="header-section-number">16.3</span> Joint analysis<a class="anchor" aria-label="anchor" href="#joint-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>The ideal use case for such functions is to plot them side by side with a regular UMAP as well:</p>
<div class="sourceCode" id="cb141"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_DimPlot.html">do_DimPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                         plot_cell_borders <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                         border.size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>                         pt.size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_CopyNumberVariantPlot.html">do_CopyNumberVariantPlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                                        infercnv_object <span class="op">=</span> <span class="va">infercnv_object</span>,</span>
<span>                                        using_metacells <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                                        metacell_mapping <span class="op">=</span> <span class="va">sample</span><span class="op">$</span><span class="va">metacell_mapping</span>,</span>
<span>                                        chromosome_locations <span class="op">=</span> <span class="va">human_chr_locations</span>,</span>
<span>                                        chromosome_focus <span class="op">=</span> <span class="st">"11"</span>,</span>
<span>                                        rotate_x_axis_labels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">`11p_umap`</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">`11p_geyser`</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span><span class="op">)</span> <span class="op">/</span> <span class="va">p3</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="21-CopyNumberVariantPlots_files/figure-html/unnamed-chunk-6-1.png" alt="SCpubr::do_CopyNumberVariantPlot joint analysis." width="100%" height="100%"><p class="caption">
Figure 1.5: SCpubr::do_CopyNumberVariantPlot joint analysis.
</p>
</div>
<p>This way, we can see which clusters are in the UMAP, see their scores and also visualize the distributions!</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="ligand-receptor-analysis.html"><span class="header-section-number">15</span> Ligand-Receptor analysis</a></div>
<div class="next"><a href="pathway-activity-inference-analysis.html"><span class="header-section-number">17</span> Pathway Activity inference analysis</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><form role="search" class="mt-4 mb-4">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>
      <div class="book-extra">
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#copy-number-variant-analysis-plots"><span class="header-section-number">16</span> Copy Number Variant analysis plots</a></li>
<li><a class="nav-link" href="#transferring-the-scores-to-a-featureplot"><span class="header-section-number">16.1</span> Transferring the scores to a FeaturePlot</a></li>
<li><a class="nav-link" href="#plotting-the-scores-grouped-by-a-group-of-interest"><span class="header-section-number">16.2</span> Plotting the scores grouped by a group of interest</a></li>
<li><a class="nav-link" href="#joint-analysis"><span class="header-section-number">16.3</span> Joint analysis</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<!-- Modified page for chapter. --><li><a id="book-source" href="https://github.com/enblacar/SCpubr-book/blob/main/21-CopyNumberVariantPlots.Rmd">View chapter source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/enblacar/SCpubr-book/edit/main/21-CopyNumberVariantPlots.Rmd">Edit this chapter <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    
  </div>

</nav>
</div>
</div> <!-- .container -->




<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</div>
</body>
<!-- Moved the footer outside of the body to make it span all page width. --><footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p class="text-center">Written by <strong>Enrique Blanco Carmona</strong> on 01 October, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p class="text-center">This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</html>
