<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>2 Feature plots | SCpubr</title>
<meta name="author" content="Enrique Blanco Carmona">
<meta name="description" content="Another flagship function in Seurat is Seurat::FeaturePlot(). It is basically the counterpart of Seurat::DimPlot() which, instead of coloring the cells based on a categorical color scale, it uses...">
<meta name="generator" content="bookdown 0.28 with bs4_book()">
<meta property="og:title" content="2 Feature plots | SCpubr">
<meta property="og:type" content="book">
<meta property="og:url" content="https://enblacar.github.io/SCpubr-book/feature-plots.html">
<meta property="og:image" content="https://enblacar.github.io/SCpubr-book/images/logo.png">
<meta property="og:description" content="Another flagship function in Seurat is Seurat::FeaturePlot(). It is basically the counterpart of Seurat::DimPlot() which, instead of coloring the cells based on a categorical color scale, it uses...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="2 Feature plots | SCpubr">
<meta name="twitter:description" content="Another flagship function in Seurat is Seurat::FeaturePlot(). It is basically the counterpart of Seurat::DimPlot() which, instead of coloring the cells based on a categorical color scale, it uses...">
<meta name="twitter:image" content="https://enblacar.github.io/SCpubr-book/images/logo.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.4.0/transition.js"></script><script src="libs/bs3compat-0.4.0/tabs.js"></script><script src="libs/bs3compat-0.4.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link rel="shortcut icon" href="images/SCpubr_logo_favicon.png">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a id="toc_title" href="index.html" title="">SCpubr</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <div class="book-extra"></div>
      <nav id="left-panel" aria-label="Table of contents"><ul class="book-toc list-unstyled">
<li><a class="" href="index.html">About this package</a></li>
<li><a class="" href="installation-guide.html">Installation guide</a></li>
<li><a class="" href="dataset-used.html">Dataset used</a></li>
<li><a class="" href="dim-plots.html"><span class="header-section-number">1</span> Dim plots</a></li>
<li><a class="active" href="feature-plots.html"><span class="header-section-number">2</span> Feature plots</a></li>
<li><a class="" href="nebulosa-plots.html"><span class="header-section-number">3</span> Nebulosa plots</a></li>
<li><a class="" href="bee-swarm-plots.html"><span class="header-section-number">4</span> Bee Swarm plots</a></li>
<li><a class="" href="violin-plots.html"><span class="header-section-number">5</span> Violin plots</a></li>
<li><a class="" href="ridge-plots.html"><span class="header-section-number">6</span> Ridge plots</a></li>
<li><a class="" href="dot-plots.html"><span class="header-section-number">7</span> Dot plots</a></li>
<li><a class="" href="bar-plots.html"><span class="header-section-number">8</span> Bar plots</a></li>
<li><a class="" href="box-plots.html"><span class="header-section-number">9</span> Box plots</a></li>
<li><a class="" href="geyser-plots.html"><span class="header-section-number">10</span> Geyser plots</a></li>
<li><a class="" href="sankey-and-alluvial-plots.html"><span class="header-section-number">11</span> Sankey and alluvial plots</a></li>
<li><a class="" href="chord-diagram-plots.html"><span class="header-section-number">12</span> Chord Diagram plots</a></li>
<li><a class="" href="volcano-plots.html"><span class="header-section-number">13</span> Volcano plots</a></li>
<li><a class="" href="group-wise-de-analysis-plots.html"><span class="header-section-number">14</span> Group-wise DE analysis plots</a></li>
<li><a class="" href="term-enrichment-plots.html"><span class="header-section-number">15</span> Term Enrichment Plots</a></li>
<li><a class="" href="enrichment-score-heatmaps.html"><span class="header-section-number">16</span> Enrichment score heatmaps</a></li>
<li><a class="" href="correlation-matrix-heatmaps.html"><span class="header-section-number">17</span> Correlation matrix heatmaps</a></li>
<li><a class="" href="cellular-state-plots.html"><span class="header-section-number">18</span> Cellular State Plots</a></li>
<li><a class="" href="ligand-receptor-analysis.html"><span class="header-section-number">19</span> Ligand-Receptor analysis</a></li>
<li><a class="" href="copy-number-variant-analysis-plots.html"><span class="header-section-number">20</span> Copy Number Variant analysis plots</a></li>
<li><a class="" href="pathway-activity-inference-analysis.html"><span class="header-section-number">21</span> Pathway Activity inference analysis</a></li>
<li><a class="" href="tf-activity-inference-analysis.html"><span class="header-section-number">22</span> TF Activity inference analysis</a></li>
<li><a class="" href="pseudotime-analysis.html"><span class="header-section-number">23</span> Pseudotime analysis</a></li>
<li><a class="" href="color-palettes.html"><span class="header-section-number">24</span> Color palettes</a></li>
<li><a class="" href="common-features-across-plots.html">Common features across plots</a></li>
<li><a class="" href="save-the-figures.html">Save the figures</a></li>
<li><a class="" href="citation.html">Citation</a></li>
<li><a class="" href="news.html">News</a></li>
<li><a class="" href="acknowledgements.html">Acknowledgements</a></li>
</ul>

        <div class="book-extra">
          <!-- Duplicated this entry to point to the package repo. -->
          <p class="mb-2"><a id="package-repo" href="https://github.com/enblacar/SCpubr">View package source <i class="fab fa-github"></i></a></p>
          <p><a id="book-repo" href="https://github.com/enblacar/SCpubr-book">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="feature-plots" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Feature plots<a class="anchor" aria-label="anchor" href="#feature-plots"><i class="fas fa-link"></i></a>
</h1>
<p>Another flagship function in <code>Seurat</code> is <code><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">Seurat::FeaturePlot()</a></code>. It is basically the counterpart of <code><a href="https://satijalab.org/seurat/reference/DimPlot.html">Seurat::DimPlot()</a></code> which, instead of coloring the cells based on a categorical color scale, it uses a continuous scale instead, according to a variable provided by the user. This can range from gene expression, to metadata variables such as the number of genes, or even values such as a principal component.</p>
<div id="basic-usage-1" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Basic usage<a class="anchor" aria-label="anchor" href="#basic-usage-1"><i class="fas fa-link"></i></a>
</h2>
<p>This is SCpubr’s take on <code><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">Seurat::FeaturePlot()</a></code>:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Seurat's Feature Plot.</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                          features <span class="op">=</span> <span class="st">"PC_1"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># SCpubr's Feature Plot.</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>,</span>
<span>                             features <span class="op">=</span> <span class="st">"PC_1"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-2"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-2-1.png" alt="SCpubr do_FeaturePlot vs Seurat FeaturePlot" width="100%" height="100%"><p class="caption">
Figure 1.1: SCpubr do_FeaturePlot vs Seurat FeaturePlot
</p>
</div>
<p>By default, <code><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">SCpubr::do_FeaturePlot()</a></code> enlarges the dots a bit and modifies the legend, to better stand out. Even though axes are removed from UMAP reductions, the title for the axes is kept for any other reduction used. The same is applied for UMAP if the default dimension order is altered.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use case with PCA embedding.</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>                             features <span class="op">=</span> <span class="st">"CD14"</span>,</span>
<span>                             plot.title <span class="op">=</span> <span class="st">"Plotting PCA coordinates"</span>,</span>
<span>                             reduction <span class="op">=</span> <span class="st">"pca"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use case with non-canonical dimensions.                             </span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>                             features <span class="op">=</span> <span class="st">"CD14"</span>,</span>
<span>                             plot.title <span class="op">=</span> <span class="st">"Plotting UMAP coordinates in a different order"</span>,</span>
<span>                             dims <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-3"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-3-1.png" alt="SCpubr Feature showing axes titles." width="100%" height="100%"><p class="caption">
Figure 1.2: SCpubr Feature showing axes titles.
</p>
</div>
</div>
<div id="multiple-features" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Multiple features<a class="anchor" aria-label="anchor" href="#multiple-features"><i class="fas fa-link"></i></a>
</h2>
<p><code><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">SCpubr::do_FeaturePlot()</a></code> can tackle multiple queries at the same time. Plots can be rearranged as desired with <code>ncol</code> parameter:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Query multiple features at the same time.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nCount_RNA"</span>, </span>
<span>                                                 <span class="st">"nFeature_RNA"</span>, </span>
<span>                                                 <span class="st">"percent.mt"</span>, </span>
<span>                                                 <span class="st">"CD14"</span><span class="op">)</span>, </span>
<span>                            plot.title <span class="op">=</span> <span class="st">"A collection of features"</span>, </span>
<span>                            ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-4"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-4-1.png" alt="SCpubr FeaturePlot, with 4 features and 2 columns" width="100%" height="100%"><p class="caption">
Figure 1.3: SCpubr FeaturePlot, with 4 features and 2 columns
</p>
</div>
</div>
<div id="working-with-subsets-of-cells" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Working with subsets of cells<a class="anchor" aria-label="anchor" href="#working-with-subsets-of-cells"><i class="fas fa-link"></i></a>
</h2>
<p>One of the things that can be misleading is that it could be the case that a <strong>very specific subset of cells are driving the ends of the color scales</strong>. Let’s say that we have identified that clusters 2, 5, and 8 are responsible for this behavior. We would like to plot the values without taking those cells into consideration. The very first thing that comes to mind is, indeed, removing the cells completely, resulting in the following:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset the cells in the Seurat object to remove unwanted cells.</span></span>
<span><span class="va">cells.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">sample</span><span class="op">[</span>, <span class="op">!</span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">seurat_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2"</span>, <span class="st">"5"</span>, <span class="st">"8"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span><span class="op">[</span>, <span class="va">cells.plot</span><span class="op">]</span>, </span>
<span>                            features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD14"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-5"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-5-1.png" alt="SCpubr FeaturePlot, removing cells from clusters 2, 5 and 8" width="100%" height="100%"><p class="caption">
Figure 1.4: SCpubr FeaturePlot, removing cells from clusters 2, 5 and 8
</p>
</div>
<p>However, this falls back to the same problem as with <code><a href="https://satijalab.org/seurat/reference/DimPlot.html">Seurat::DimPlot</a></code> with the <code>split.by</code> parameter. We lose the overall context of where the cells are, as we lost the original <strong>UMAP silhouette</strong>. This can be fixed by using the <code>cells.highlight</code> parameter:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using cells.highlight parameter to select the cells we want to include in the plot.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>                            cells.highlight <span class="op">=</span> <span class="va">cells.plot</span>, </span>
<span>                            features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD14"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-6-1.png" alt="SCpubr FeaturePlot, masking cells from clusters 2, 5 and 8" width="100%" height="100%"><p class="caption">
Figure 1.5: SCpubr FeaturePlot, masking cells from clusters 2, 5 and 8
</p>
</div>
<p>This way, by adding the cells back and modifying the legend, we accomplish the desired effect. Furthermore, <code><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">SCpubr::do_FeaturePlot()</a></code> also accepts a vector of the identities to plot, as long as they are part of <code>levels(seurat_object)</code>. This can be achieved by using the <code>idents.highlight</code> instead.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Selecting given identitites to include in the plot.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>                            idents.highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span>, <span class="st">"2"</span>, <span class="st">"5"</span>, <span class="st">"8"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>, </span>
<span>                            features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CD14"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-7-1.png" alt="SCpubr FeaturePlot, masking cells from clusters 2, 5 and 8 using idents.highlight" width="100%" height="100%"><p class="caption">
Figure 1.6: SCpubr FeaturePlot, masking cells from clusters 2, 5 and 8 using idents.highlight
</p>
</div>
</div>
<div id="splitting-the-featureplot-by-a-variable" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Splitting the FeaturePlot by a variable<a class="anchor" aria-label="anchor" href="#splitting-the-featureplot-by-a-variable"><i class="fas fa-link"></i></a>
</h2>
<p>In some cases, we want to visualize a given feature, let’s say, the expression of <code>LYN</code> across another variable, for instance <code>seurat_clusters</code>. This can be achieved easily in <code>Seurat</code> by using the <code>split.by</code> parameter. For the sake of representation, let’s reduce the number of clusters.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Group clusters into three values for visualization purposes.</span></span>
<span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">sample</span><span class="op">$</span><span class="va">seurat_clusters</span><span class="op">)</span></span>
<span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span><span class="op">[</span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Cluster 1"</span></span>
<span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span><span class="op">[</span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"4"</span>, <span class="st">"5"</span>, <span class="st">"6"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Cluster 2"</span></span>
<span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span><span class="op">[</span><span class="va">sample</span><span class="op">$</span><span class="va">new_clusters</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"7"</span>, <span class="st">"8"</span>, <span class="st">"9"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Cluster 3"</span></span>
<span></span>
<span><span class="co"># Seurat Feature Plot using split.by.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html">FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                         features <span class="op">=</span> <span class="st">"LYN"</span>, </span>
<span>                         split.by <span class="op">=</span> <span class="st">"new_clusters"</span><span class="op">)</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-8-1.png" alt="Seurat FeaturePlot split by clusters." width="100%" height="100%"><p class="caption">
Figure 1.7: Seurat FeaturePlot split by clusters.
</p>
</div>
<p>Apart from the general aesthetic discrepancies mentioned before, there are two main aspects missing. First, is the loss of the <strong>UMAP silhouette</strong>. Secondly, knowing that, by default, the color scale is shared across all the plots, we would like to know the range of values this color scale is displaying. In this two aspects is where <code><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">SCpubr::do_FeaturePlot()</a></code> delves into:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SCpubr Feature Plot using split.by</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span>sample <span class="op">=</span> <span class="va">sample</span>, </span>
<span>                            features <span class="op">=</span> <span class="st">"LYN"</span>, </span>
<span>                            split.by <span class="op">=</span> <span class="st">"new_clusters"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-9"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-9-1.png" alt="SCpubr FeaturePlot split by clusters." width="100%" height="100%"><p class="caption">
Figure 1.8: SCpubr FeaturePlot split by clusters.
</p>
</div>
<p>This way, we are able to achieve a better understandable plot. Furthermore, we can query multiple features. In the following example, we provide each of the three possible examples of features:</p>
<ul>
<li>
<code>LYN</code> is a gene located in the rownames of the count matrix (`rownames(sample)``).</li>
<li>
<code>nCount_RNA</code> is a metadata variable located in <code>colnames(sample@meta.data)</code>.</li>
<li>
<code>PC_1</code> is a column of a dimensional reduction embedding, located in <code>colnames(sample@reductions[["pca"]][[]])</code>
</li>
</ul>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use split.by with multiple features.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                            features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"LYN"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"PC_1"</span><span class="op">)</span>, </span>
<span>                            split.by <span class="op">=</span> <span class="st">"new_clusters"</span><span class="op">)</span></span>
<span>                            </span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-10-1.png" alt="SCpubr FeaturePlot split by clusters using multiple features." width="100%" height="100%"><p class="caption">
Figure 1.9: SCpubr FeaturePlot split by clusters using multiple features.
</p>
</div>
<p>We can even subset the previous plot to any set ofidentities in <code>split.by</code> we are particularly interested in. The color scale also limits itself to contain only the values in the selected identities. This can be achieved by provided a vector with the identities to <code>split.by.idents</code> parameter.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using split.by.idents parameter alongside split.by.</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                            features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"LYN"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"PC_1"</span><span class="op">)</span>, </span>
<span>                            split.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>, </span>
<span>                            split.by.idents <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"2"</span>, <span class="st">"5"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-11-1.png" alt="SCpubr FeaturePlot split by clusters using multiple features and selected identities." width="100%" height="100%"><p class="caption">
Figure 1.10: SCpubr FeaturePlot split by clusters using multiple features and selected identities.
</p>
</div>
</div>
<div id="apply-symmetrical-color-scales" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Apply symmetrical color scales<a class="anchor" aria-label="anchor" href="#apply-symmetrical-color-scales"><i class="fas fa-link"></i></a>
</h2>
<p>There are times in which we want to color the Feature plot using a two-end continuous color scale. We can achieve this by using <code>enforce_symmetry = TRUE</code>:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Enforce two-end symmetrical color scale.</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                             features <span class="op">=</span> <span class="st">"PC_1"</span>,</span>
<span>                             enforce_symmetry <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">SCpubr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SCpubr/man/do_FeaturePlot.html">do_FeaturePlot</a></span><span class="op">(</span><span class="va">sample</span>, </span>
<span>                             features <span class="op">=</span> <span class="st">"PC_1"</span>,</span>
<span>                             enforce_symmetry <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span></span>
<span><span class="va">p</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="04-FeaturePlots_files/figure-html/unnamed-chunk-12-1.png" alt="SCpubr FeaturePlot using a two-end continuous scale." width="100%" height="100%"><p class="caption">
Figure 1.11: SCpubr FeaturePlot using a two-end continuous scale.
</p>
</div>
<p>As one can observe, this parameters sets a scale centered around 0, with the two ends being of the same value. This allows for direct comparison between the colors.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="dim-plots.html"><span class="header-section-number">1</span> Dim plots</a></div>
<div class="next"><a href="nebulosa-plots.html"><span class="header-section-number">3</span> Nebulosa plots</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><form role="search" class="mt-4 mb-4">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>
      <div class="book-extra">
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#feature-plots"><span class="header-section-number">2</span> Feature plots</a></li>
<li><a class="nav-link" href="#basic-usage-1"><span class="header-section-number">2.1</span> Basic usage</a></li>
<li><a class="nav-link" href="#multiple-features"><span class="header-section-number">2.2</span> Multiple features</a></li>
<li><a class="nav-link" href="#working-with-subsets-of-cells"><span class="header-section-number">2.3</span> Working with subsets of cells</a></li>
<li><a class="nav-link" href="#splitting-the-featureplot-by-a-variable"><span class="header-section-number">2.4</span> Splitting the FeaturePlot by a variable</a></li>
<li><a class="nav-link" href="#apply-symmetrical-color-scales"><span class="header-section-number">2.5</span> Apply symmetrical color scales</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<!-- Modified page for chapter. --><li><a id="book-source" href="https://github.com/enblacar/SCpubr-book/blob/main/04-FeaturePlots.Rmd">View chapter source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/enblacar/SCpubr-book/edit/main/04-FeaturePlots.Rmd">Edit this chapter <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    
  </div>

</nav>
</div>
</div> <!-- .container -->




</div>
</body>
<!-- Moved the footer outside of the body to make it span all page width. --><footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p class="text-center">Written by <strong>Enrique Blanco Carmona</strong> on 17 October, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p class="text-center">This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</html>
